atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/sweep/queue/KvsSweepQueueTest.java

```diff
@@ -0,0 +1,243 @@
+        assertValue(TABLE_CONSERVATIVE, TS + 1, TS);
+        assertTombstone(TABLE_CONSERVATIVE, TS2 + 1, TS2);
+
+        sweepQueue.sweepNextBatch(ShardAndStrategy.conservative(CONS_SHARD));
+        assertSentinel(TABLE_CONSERVATIVE, TS + 1);
+        assertTombstone(TABLE_CONSERVATIVE, TS2 + 1, TS2);
+    }
+
+    @Test
+    public void thoroughSweepDeletesTombstoneIfLatestWrite() {
+        sweepQueue.enqueue(tombstoneToDefaultCell(TABLE_THOROUGH, TS), TS);
+        sweepQueue.enqueue(tombstoneToDefaultCell(TABLE_THOROUGH, TS2), TS2);
+        assertTombstone(TABLE_THOROUGH, TS + 1, TS);
+        assertTombstone(TABLE_THOROUGH, TS2 + 1, TS2);
+
+        sweepQueue.sweepNextBatch(ShardAndStrategy.thorough(THOR_SHARD));
+        assertNothing(TABLE_THOROUGH, TS + 1);
+        assertNothing(TABLE_THOROUGH, TS2 + 1);
+    }
+
+    @Test
+    public void sweepDeletesTombstonesWhenWriteHasHigherTimestamp() {
+        sweepQueue.enqueue(tombstoneToDefaultCell(TABLE_CONSERVATIVE, TS), TS);
+        sweepQueue.enqueue(writeToDefaultCell(TABLE_CONSERVATIVE, TS2), TS2);
+        assertTombstone(TABLE_CONSERVATIVE, TS + 1, TS);
+        assertValue(TABLE_CONSERVATIVE, TS2 + 1, TS2);
+
+        sweepQueue.sweepNextBatch(ShardAndStrategy.conservative(CONS_SHARD));
+        assertSentinel(TABLE_CONSERVATIVE, TS + 1);
+        assertValue(TABLE_CONSERVATIVE, TS2 + 1, TS2);
+    }
+
+    private Map<TableReference, ? extends Map<Cell, byte[]>> writeToDefaultCell(TableReference tableRef, long ts) {
+        Map<Cell, byte[]> singleWrite = ImmutableMap.of(DEFAULT_CELL, PtBytes.toBytes(ts));
+        kvs.put(tableRef, singleWrite, ts);
+        return ImmutableMap.of(tableRef, singleWrite);
+    }
+
+    private Map<TableReference, ? extends Map<Cell, byte[]>> tombstoneToDefaultCell(TableReference tableRef, long ts) {
+        Map<Cell, byte[]> singleWrite = ImmutableMap.of(DEFAULT_CELL, PtBytes.EMPTY_BYTE_ARRAY);
+        kvs.put(tableRef, singleWrite, ts);
+        return ImmutableMap.of(tableRef, singleWrite);
+    }
+
+    private void assertValue(TableReference tableRef, long ts, long value) {
+        assertThat(readValueFromDefaultCell(tableRef, ts)).isEqualTo(value);
+    }
+
+    private long readValueFromDefaultCell(TableReference tableRef, long ts) {
+        return PtBytes.toLong(Iterables.getOnlyElement(readFromDefaultCell(tableRef, ts).values()).getContents());
+    }
+
+    private void assertSentinel(TableReference tableRef, long ts) {
+        assertTombstone(tableRef, ts, -1L);
+    }
+
+    private void assertTombstone(TableReference tableRef, long readTs, long tombTs) {
+        Value readValue = Iterables.getOnlyElement(readFromDefaultCell(tableRef, readTs).values());
+        assertThat(readValue.getTimestamp()).isEqualTo(tombTs);
+        assertThat(readValue.getContents()).isEmpty();
+    }
+
+    private void assertNothing(TableReference tableRef, long ts) {
```

Comment:
Slightly more descriptive method names for the `assert...` methods would help in understanding what they're checking.

Reply:
Better?